Pytest | Unprintable RunAnsibleModuleFail object

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/Azure/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**
<!--
Briefly describe the problem you are having in a few paragraphs.
-->
While using pytest framework sometimes it happens that `RunAnsibleModuleFail` exception can't be handled properly:
```
self = <common.devices.PTFHost object at 0x7f95070c0550> 
module_args = ('ptf --test-dir ptftests dhcp_relay_test.DHCPTest --platform-dir ptftests --platform remote -t \'num_dhcp_servers=4;l...elay_iface_mac=\'"\'"\'b8:59:9f:a8:0b:00\'"\'"\'\' --relax --debug info --log-file /tmp/dhcp_relay_test.DHCPTest.log',) 
complex_args = {'chdir': '/root'}, module_ignore_errors = False 
module_async = False 
res = {'stderr_lines': [u'dhcp_relay_test.DHCPTest ... FAIL', u'', u'===============...': True, u'stdin': None}}, 'stdout_lines': [], u'msg': u'non-zero return code'} 
    def _run(self, *module_args, **complex_args): 
        module_ignore_errors = complex_args.pop('module_ignore_errors', False) 
        module_async = complex_args.pop('module_async', False) 
        if module_async: 
            def run_module(module_args, complex_args): 
                return self.module(*module_args, **complex_args)[self.hostname] 
            pool = ThreadPool() 
            result = pool.apply_async(run_module, (module_args, complex_args)) 
            return pool, result 
        res = self.module(*module_args, **complex_args)[self.hostname] 
        if res.is_failed and not module_ignore_errors: 
>           raise RunAnsibleModuleFail("run module {} failed".format(self.module_name), res) 
E           RunAnsibleModuleFail: <unprintable RunAnsibleModuleFail object> 
```
Looks like this piece of code introduces a bug:
```
class AnsibleHostBase(object):
    """
    @summary: The base class for various objects.

    This class filters an object from the ansible_adhoc fixture by hostname. The object can be considered as an
    ansible host object although it is not under the hood. Anyway, we can use this object to run ansible module
    on the host.
    """
    def _run(self, *module_args, **complex_args):
        module_ignore_errors = complex_args.pop('module_ignore_errors', False)
        module_async = complex_args.pop('module_async', False)

        if module_async:
            def run_module(module_args, complex_args):
                return self.module(*module_args, **complex_args)[self.hostname]
            pool = ThreadPool()
            result = pool.apply_async(run_module, (module_args, complex_args))
            return pool, result

        res = self.module(*module_args, **complex_args)[self.hostname]
        if res.is_failed and not module_ignore_errors:
            raise RunAnsibleModuleFail("run module {} failed".format(self.module_name), res)
```

Original `AnsibleError` implementation:
```
class AnsibleError(Exception):
    '''
    This is the base class for all errors raised from Ansible code,
    and can be instantiated with two optional parameters beyond the
    error message to control whether detailed information is displayed
    when the error occurred while parsing a data file of some kind.
    Usage:
        raise AnsibleError('some message here', obj=obj, show_content=True)
    Where "obj" is some subclass of ansible.parsing.yaml.objects.AnsibleBaseYAMLObject,
    which should be returned by the DataLoader() class.
    '''
    def __str__(self):
        return self.message

    def __repr__(self):
        return self.message
```

Pytest `RunAnsibleModuleFail` implementation:
```
class RunAnsibleModuleFail(AnsibleError):

    """Sub-class AnsibleError when module exceptions occur."""

    def __init__(self, msg, results=None):
        super(RunAnsibleModuleFail, self).__init__(msg)
        self.results = results

    def __str__(self):
        return "{}\nAnsible Results => {}".format(self.message, dump_ansible_results(self.results))

```

**Steps to reproduce the issue:**
* N/A

**Describe the results you received:**
* We have got module exception as unprintable object

**Describe the results you expected:**
* We expect that module exception will be converted to a string with the meaningful definition
* `__str__` or `__repr__` should have internal exception protection

**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
* N/A

--- a/tests/common/errors.py
+++ b/tests/common/errors.py
@@ -1,17 +1,20 @@
 """
 Customize exceptions
 """
-from ansible.plugins.loader import callback_loader
 from ansible.errors import AnsibleError
+from ansible.plugins.loader import callback_loader
 
 
 class UnsupportedAnsibleModule(Exception):
     pass
 
 
-def dump_ansible_results(results, stdout_callback='yaml'):
-    cb = callback_loader.get(stdout_callback)
-    return cb._dump_results(results) if cb else results
+def dump_ansible_results(results, stdout_callback='json'):
+    try:
+        cb = callback_loader.get(stdout_callback)
+        return cb._dump_results(results) if cb else results
+    except Exception:
+        return str(results)
 
 
 class RunAnsibleModuleFail(AnsibleError):
@@ -22,8 +25,15 @@ class RunAnsibleModuleFail(AnsibleError):
         super(RunAnsibleModuleFail, self).__init__(msg)
         self.results = results
 
+    def _to_string(self):
+        return "{}, Ansible Results =>\n{}".format(self.message, dump_ansible_results(self.results))
+
     def __str__(self):
-        return "{}\nAnsible Results => {}".format(self.message, dump_ansible_results(self.results))
+        return self._to_string()
+
+    def __repr__(self):
+        return self._to_string()
+
 
 class MissingInputError(Exception):
     pass

